src/meta/ConfigManager.h

```diff
@@ -0,0 +1,200 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef META_CONFIGMANAGER_H_
+#define META_CONFIGMANAGER_H_
+
+#include "base/Base.h"
+#include <gtest/gtest_prod.h>
+#include "meta/client/MetaClient.h"
+
+namespace nebula {
+namespace meta {
+
+class ConfigManager {
+    FRIEND_TEST(ConfigManTest, MetaConfigManTest);
+    FRIEND_TEST(ConfigManTest, MemConfigManagerTest);
+
+public:
+    static std::unique_ptr<ConfigManager> instance(MetaClient* client = nullptr);
+
+    virtual ~ConfigManager();
+
+    virtual folly::Future<StatusOr<bool>> setConfig(folly::StringPiece space,
+                                                    folly::StringPiece module,
+                                                    folly::StringPiece name,
+                                                    const VariantType &value,
+                                                    bool checkRegistered = true);
+
+    virtual folly::Future<StatusOr<ConfigItem>> getConfigAsInt64(folly::StringPiece space,
+                                                                 folly::StringPiece module,
+                                                                 folly::StringPiece name) = 0;
+
+    virtual folly::Future<StatusOr<ConfigItem>> getConfigAsDouble(folly::StringPiece space,
+                                                                  folly::StringPiece module,
+                                                                  folly::StringPiece name) = 0;
+
+    virtual folly::Future<StatusOr<ConfigItem>> getConfigAsBool(folly::StringPiece space,
+                                                                folly::StringPiece module,
+                                                                folly::StringPiece name) = 0;
+
+    virtual folly::Future<StatusOr<ConfigItem>> getConfigAsString(folly::StringPiece space,
+                                                                  folly::StringPiece module,
+                                                                  folly::StringPiece name) = 0;
+
+    virtual folly::Future<StatusOr<std::vector<ConfigItem>>> listConfigs(
+            folly::StringPiece space, folly::StringPiece module) = 0;
+
+
+    virtual folly::Future<StatusOr<bool>> registerConfig(const std::string& space,
+                                                         const std::string& module,
+                                                         const std::string& name,
+                                                         const VariantTypeEnum& type,
+                                                         const VariantType& defaultValue);
+
+protected:
+    virtual folly::Future<StatusOr<bool>> setConfigAsInt64(folly::StringPiece space,
```

Comment:
Why make the methods to be protected?

Reply:
We only use the setConfig method in ConfigManger for console now,  which will call the Int64 and other method.